src/dialogs/more-info/components/ha-more-info-toggle.ts

```diff
@@ -0,0 +1,149 @@
+import { mdiFlash, mdiFlashOff } from "@mdi/js";
+import { HassEntity } from "home-assistant-js-websocket";
+import { css, CSSResultGroup, html, LitElement, TemplateResult } from "lit";
+import { customElement, property } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { styleMap } from "lit/directives/style-map";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { stateActive } from "../../../common/entity/state_active";
+import { stateColorCss } from "../../../common/entity/state_color";
+import "../../../components/ha-control-button";
+import "../../../components/ha-control-switch";
+import { UNAVAILABLE, UNKNOWN } from "../../../data/entity";
+import { HomeAssistant } from "../../../types";
+
+@customElement("ha-more-info-toggle")
+export class HaMoreInfoToggle extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj!: HassEntity;
+
+  @property({ attribute: false }) public iconPathOn?: string;
+
+  @property({ attribute: false }) public iconPathOff?: string;
+
+  private _valueChanged(ev) {
+    const checked = ev.target.checked as boolean;
+
+    if (checked) {
+      this._turnOn();
+    } else {
+      this._turnOff();
+    }
+  }
+
+  private _turnOn() {
+    const domain = computeDomain(this.stateObj!.entity_id);
+    this.hass.callService(domain, "turn_on", {
+      entity_id: this.stateObj!.entity_id,
+    });
+  }
+
+  private _turnOff() {
+    const domain = computeDomain(this.stateObj!.entity_id);
+    this.hass.callService(domain, "turn_off", {
+      entity_id: this.stateObj!.entity_id,
+    });
+  }
```

Comment:
For the future, we should probably align with <https://github.com/home-assistant/frontend/blob/dev/src/components/entity/ha-entity-toggle.ts#L116>

Reply:
I don't think `cover` and `lock` will use this component because we will have a dedicated more info with multiple buttons.